Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -966,6 +1019,71 @@ theorem exists_boundary_dart {u v : V} (p : G.Walk u v) (S : Set V) (uS : u ∈
       exact ⟨d, List.Mem.tail _ hd, hcd⟩
     · exact ⟨⟨(x, y), a⟩, List.Mem.head _, uS, h⟩
 
+lemma getVert_copy  {u v w x: V} (p : G.Walk u v) (i : ℕ) (h : u = w) (h' : v = x) :
+    (p.copy h h').getVert i = p.getVert i := by
+  cases p with
+  | nil =>
+    rw [getVert_of_length_le _ (by simp only [length_nil, Nat.zero_le] : nil.length ≤ _)]
+    rw [getVert_of_length_le _ (by simp only [length_copy, length_nil, Nat.zero_le])]
+    exact h'.symm
+  | cons hadj q =>
+    by_cases hi : i = 0
+    · simp only [hi, getVert_zero, h.symm]
+    · simp only [copy_cons, cons_getVert_succ]
+      rw [cons_getVert _ _ hi]
+      rw [cons_getVert _ _ hi]
+      apply getVert_copy
+
+lemma getVert_tail {u v n} (p : G.Walk u v) (hnp: ¬ p.Nil) :
+    p.tail.getVert n = p.getVert (n + 1) := by
+  match p with
+  | .nil => rfl
+  | .cons h q =>
+    simp only [cons_getVert_succ, tail_cons_eq, cons_getVert]
+    exact getVert_copy q n (getVert_zero q).symm rfl
+
+lemma cons_getVert_one (q : G.Walk v w) (hadj : G.Adj u v) :
```

Comment:
```suggestion
@[simp] lemma getVert_ cons_one (q : G.Walk v w) (hadj : G.Adj u v) :
```

Reply:
This one is a consequence of cons_getVert_succ and getVert_zero. I couldn't figure out how to add it without the linter shouting at me, am I missing something?

Reply:
Oh the linter is right here. Forget me :)

Reply:
Well actually please do
```suggestion
lemma getVert_cons_one (q : G.Walk v w) (hadj : G.Adj u v) :
```
still

Reply:
Fixed along with some other misnames:)